### DISCUSSION TITLE
there's no public/packs directory

### DISCUSSION THREAD

**Discussion by yinnxinn** [mastodon/mastodon#22114] [Q&A]
### Steps to reproduce the problem

 I follow the[ instructions](https://docs.joinmastodon.org/admin/install/) in the official web . And after start the service , I get 404 when get static files in public/packs . 
![image](https://user-images.githubusercontent.com/23116748/206075246-986095a9-73dc-43da-badf-6f73b4acb096.png)

Compared with  #20535  issue, I have no  `Running RAILS_ENV=production rails assets:precompile ...` process in my log . 

> 
> Your instance is identified by its domain name. Changing it afterward will break things.
> Domain name:
> Do you want to enable single user mode? (y/N)
> Are you using Docker to run Mastodon? (Y/n)
> 
> PostgreSQL host: db
> PostgreSQL port: 5432
> Name of PostgreSQL database: postgres
> Name of PostgreSQL user: postgres
> Password of PostgreSQL user:
> Database configuration works! 
> 
> Redis host: redis
> Redis port: 6379
> Redis password:
> Redis configuration works! 
> 
> Do you want to store uploaded files on the cloud? (y/N)
> 
> Do you want to send e-mails from localhost? (y/N)
> 
> Do you want to send e-mails from localhost? No
> SMTP server: smtp.126.com
> SMTP port: 25
> SMTP username: @126.com
> SMTP password:
> SMTP authentication: plain
> SMTP OpenSSL verify mode: none
> E-mail address to send e-mails "from": @126.com
> Send a test e-mail with this configuration right now? Yes
> Send test e-mail to: @qq.com
> 
> This configuration will be written to .env.production
> Save configuration? Yes
> Below is your configuration, save it to an .env.production file outside Docker:
> 
> Now that configuration is saved, the database schema must be loaded.
> If the database already exists, this will erase its contents.
> Prepare the database now? Yes
> Running RAILS_ENV=production rails db:setup ...
> 
> Database 'postgres' already exists
> Error connecting to Redis on localhost:6379 (Errno::ECONNREFUSED)
> Error connecting to Redis on localhost:6379 (Errno::ECONNREFUSED)
> Switching object-storage-safely from green to red because Redis::CannotConnectError Error connecting to Redis on localhost:6379 (Errno::ECONNREFUSED)
> Error connecting to Redis on localhost:6379 (Errno::ECONNREFUSED)
> Done!
> 
> All done! You can now power on the Mastodon server ?
> 
> Do you want to create an admin user straight away? (Y/n)
> Username:
> E-mail:
> Error connecting to Redis on localhost:6379 (Errno::ECONNREFUSED)
> Error connecting to Redis on localhost:6379 (Errno::ECONNREFUSED)
> Switching object-storage-safely from green to red because Redis::CannotConnectError Error connecting to Redis on localhost:6379 (Errno::ECONNREFUSED)
> Error connecting to Redis on localhost:6379 (Errno::ECONNREFUSED)
> You can login with the password: xxxxxxxxxxxxxxx
> You can change your password once you login.`

When I run assets:precompile alone , I get log as follow
![image](https://user-images.githubusercontent.com/23116748/206074372-9c99664b-c6e6-4cd9-b742-9e64d8053a91.png)

there's no packs files generated as #20535. 

### Expected behaviour

website normally

### Actual behaviour

get many 404 error

### Specifications

Mastodon V4.0.2
Chrome

**Comment by NicoBerlo** [comment]
I'm getting the same error with v4.1.2
I managed to copy the packs dir from another install where it worked.
Anyone could help ?

**Comment by yinnxinn** [comment]
邮件已收到，谢谢

**Comment by renchap** [comment]
The `assets:precompile` task should generate those.

If it did not run correctly, you can re-run it using: `RAILS_ENV=production bundle exec rails assets:precompile`
